### PR TITLE
Update BMC Software

### PR DESCRIPTION
**Pull request by DFFspace**

### CODE REVIEW

File: software/README.md [hunk 1]
```diff
@@ -297,7 +297,7 @@ _Note: daily releases of this software list are listed, including CSV and JSON f
...
 | BMC Software  | Application Restart Control for IMS    | | Not vuln | Not vuln | Not vuln | Not vuln |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
 | BMC Software  | Application Restart Control for VSAM    | | Not vuln | Not vuln | Not vuln | Not vuln |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
-| BMC Software  | Bladelogic Database Automation   | | Not vuln | Fix |  | | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=Bladelogic%20Database%20Automation) |
+| BMC Software  | Bladelogic Database Automation   | 8.9.03, 20.19.01, 20.19.02, 20.19.03, 20.20.01, 20.20.02 | Not vuln | Fix | Fix | Vulnerable | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=Bladelogic%20Database%20Automation) |
```
Comment (tschmidtb51) [inline software/README.md:300]: According from to the information, I retrieved from [here](https://community.bmc.com/sfc/servlet.shepherd/document/download/0693n00000FSLXJAA5?operationContext=S1) there is:
- no statement about status of CVE-2021-45105
- No fix available for 8.9.03

Suggestion: Split lines to indicate that 8.9.03 is vulnerable

File: software/README.md [hunk 2]
```diff
@@ -312,25 +312,25 @@ _Note: daily releases of this software list are listed, including CSV and JSON f
...
-| BMC Software  | BMC AMI Ops Insight | | Not vuln | Fix |  | | |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=Dec%2020%2C%202021-,BMC%20AMI%20Ops%20Insight,-Fix%20Available%20%E2%80%93%20PTF) |
+| BMC Software  | BMC AMI Ops Infrastructure (MVI) - CRA component | 7.0 | Not vuln | Fix | Fix | Vulnerable | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=BMC%20AMI%20Ops%20Infrastructure%20(MVI)%20%2D%20CRA) |
+| BMC Software  | BMC AMI Ops Insight | 1.0, 1.2 | Not vuln | Fix | Fix | Vulnerable | |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=Dec%2020%2C%202021-,BMC%20AMI%20Ops%20Insight,-Fix%20Available%20%E2%80%93%20PTF) |
```
Comment (tschmidtb51) [inline software/README.md:318]: Same here: There seems to be no fix for 1.0

File: software/README.md [hunk 3]
```diff
@@ -373,45 +373,48 @@ _Note: daily releases of this software list are listed, including CSV and JSON f
...
-| BMC Software  | MainView Middleware Monitor | |  | Vulnerable |  | | Fix expected on Dec 20th |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
+| BMC Software  | MainView Middleware Monitor | 9.1 | Not vuln | Fix | Fix | Vulnerable | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
 | BMC Software  | MainView Transaction Analyzer    | | Not vuln | Not vuln | Not vuln | Not vuln |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
 | BMC Software  | PATROL Agent    | | Not vuln | Not vuln | Not vuln | Not vuln |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
+| BMC Software  | Patrol for Sybase & Patrol for Linux KM | Sybase - 20.02, Linux - 20.05 | Not vuln | Fix | Fix | Vulnerable |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
```
Comment (tschmidtb51) [inline software/README.md:397]: The version should list the fixes which are:
- Sybase - 20.02.02 
- Linux - 20.05.01

Moreover, split lines to improve readability.

File: software/README.md [hunk 4]
```diff
@@ -373,45 +373,48 @@ _Note: daily releases of this software list are listed, including CSV and JSON f
...
-| BMC Software  | TrueSight Automation for Networks | | Not vuln | Fix |  | | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=TrueSight%20Automation%20for%20Networks) |
-| BMC Software  | TrueSight Automation for Servers - Data Warehouse | |  | Vulnerable |  | | Fix expected on Dec 17th |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
-| BMC Software  | TrueSight Automation for Servers | |  | Vulnerable |  | | Fix expected on Dec 17th |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
+| BMC Software  | TrueSight Automation Console | 19.1, 20.02, 20.02.01, 20.08.00, 20.08.01, 21.02.00, 21.02.01, 21.3 | Not vuln | Fix | Fix | Vulnerable | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
```
Comment (tschmidtb51) [inline software/README.md:403]: There is no fix for 19.1.

Please, split lines as suggested above.

File: software/README.md [hunk 5]
```diff
@@ -373,45 +373,48 @@ _Note: daily releases of this software list are listed, including CSV and JSON f
...
+| BMC Software  | TrueSight Operations Management: App Visibility Manager | 11.3.03 | Not vuln | Fix | Fix | Vulnerable | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
+| BMC Software  | TrueSight Operations Management: Infrastructure Management | 11.3.05 | Not vuln | Fix | Fix | Vulnerable | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
+| BMC Software  | TrueSight Operations Management: IT Data Analytics | 11.3.02 | Not vuln | Fix | Fix | Vulnerable | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
+| BMC Software  | TrueSight Operations Management: Presentation Server | 11.3.02,11.3.03, 11.3.04, 11.3.05 | Not vuln | Fix | Fix | Vulnerable | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
```
Comment (tschmidtb51) [inline software/README.md:411]: No fix for 11.3.02,11.3.03.

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Do you have specific information about the status according CVE-2021-45105? I didn't saw them in the document.

Please check the versions again as something got mixed between the affected and fixed ones.

File: software/README.md [hunk 6]
```diff
@@ -297,7 +297,7 @@ _Note: daily releases of this software list are listed, including CSV and JSON f
...
 | BMC Software  | Application Restart Control for IMS    | | Not vuln | Not vuln | Not vuln | Not vuln |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
 | BMC Software  | Application Restart Control for VSAM    | | Not vuln | Not vuln | Not vuln | Not vuln |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
-| BMC Software  | Bladelogic Database Automation   | | Not vuln | Fix |  | | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=Bladelogic%20Database%20Automation) |
+| BMC Software  | Bladelogic Database Automation   | 20.19.01, 20.19.02, 20.19.03, 20.20.01, 20.20.02 | Not vuln | Fix | Fix | Investigation | Version 8.9.03 not fixed | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=Bladelogic%20Database%20Automation) |
```
Comment (tschmidtb51) [inline software/README.md:300]:
```suggestion
| BMC Software  | Bladelogic Database Automation   | 8.9.03 | Not vuln | Vulnerable | Vulnerable | Investigation | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=Bladelogic%20Database%20Automation) |
| BMC Software | Bladelogic Database Automation   | 20.19.01, 20.19.02, 20.19.03, 20.20.01, 20.20.02 | Not vuln | Workaround | Workaround | Investigation | Hotfix is available | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability#:~:text=Bladelogic%20Database%20Automation) |

File: software/README.md [hunk 7]
```diff
@@ -373,45 +373,49 @@ _Note: daily releases of this software list are listed, including CSV and JSON f
...
-| BMC Software  | MainView Middleware Monitor | |  | Vulnerable |  | | Fix expected on Dec 20th |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
+| BMC Software  | MainView Middleware Monitor | 9.1 | Not vuln | Fix | Fix | Investigation | | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
 | BMC Software  | MainView Transaction Analyzer    | | Not vuln | Not vuln | Not vuln | Not vuln |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
 | BMC Software  | PATROL Agent    | | Not vuln | Not vuln | Not vuln | Not vuln |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
+| BMC Software  | Patrol for Sybase & Patrol for Linux KM | Linux - 20.05 | Not vuln | Fix | Fix | Investigation |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
```
Comment (tschmidtb51) [inline software/README.md:397]:
```suggestion
| BMC Software  | Patrol for Linux KM | 20.05.01 | Not vuln | Fix | Fix | Investigation |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |

File: software/README.md [hunk 8]
```diff
@@ -373,45 +373,49 @@ _Note: daily releases of this software list are listed, including CSV and JSON f
...
+| BMC Software  | Patrol for Sybase & Patrol for Linux KM | Sybase - 20.02 | Not vuln | Fix | Fix | Investigation |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |
```
Comment (tschmidtb51) [inline software/README.md:398]:
```suggestion
| BMC Software  | Patrol for Sybase |  20.02.02 | Not vuln | Fix | Fix | Investigation |  |[source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-log4shell-vulnerability) |

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Thank you for the contribution.

We had a discussion today and think it is the best to use _Workaround_ if there is a hotfix available. Reasoning for this: If the version string does not change and the asset owners just sees the status _Fix_ he might believe to be safe but is vulnerable instead as he hasn't installed the hotfix. _Fix_ should mean: If you have that version - no further action required (also not installing a hotfix). That makes it easier to determine whether you are affected or not.

We are also working on documenting that rule in #710.

I also added a suggestion for the versions that do not have a hotfix available.

**Comment by DFFspace** [comment]
I have redone the whole list of BMC products and followed the How To PR review guide

- First I changed the BMC Software to just BMC
- Second I have taken out the name BMC in front of the product names to have all products alphabetically sorted
- Third I have made splits for products that have some versions hot fixed and other versions not, should make it stand out for people to see what version is hot fixed and which one not

Let me know if I missed/ forgot something

File: software/README.md [hunk 9]
```diff
@@ -293,128 +293,271 @@ _Note: daily releases of this software list are listed, including CSV and JSON f
...
+| BMC | Unload Plus for Db2 | All | Not vuln | Not vuln | Not vuln | Not vuln |  | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability) |
+| BMC | UNLOAD PLUS for IMS | All | Not vuln | Not vuln | Not vuln | Not vuln |  | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability) |
+| BMC | UNLOAD PLUS/EP for IMS | All | Not vuln | Not vuln | Not vuln | Not vuln |  | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability) |
+| BMC | User Interface Middleware (UIM) | All | Not vuln | Not vuln | Not vuln | Not vuln |  | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability) |
+| BMC | User Interface Middleware (UIM) | All | Not vuln | Not vuln | Not vuln | Not vuln |  | [source](https://community.bmc.com/s/news/aA33n000000TSUdCAO/bmc-security-advisory-for-cve202144228-cve202145046-log4shell-vulnerability) |
```
Comment (tschmidtb51) [inline software/README.md:557]: That seems to be a true duplicate. Please delete that line.
```suggestion

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Thank you for your contribution and the great effort. I have some remarks.

It looks like for those product which are listed as "Not vuln" against CVE-2021-44228 and CVE-2021-45046 the status for CVE-2021-4104 is not clear. (At least, I found some comments about the CMDB using Log4j 1.2...) Therefore, I suggest to put the in state investigation (this would be in line with the statement:
> The log4j version 1 can be vulnerable if the JNDI lookups are enabled. The BMC R&D product teams are reviewing the configuration of products using this version of log4j to ensure they are not at risk.

Note: This does **NOT** affect the _Not vuln_ status for products that are found vulnerable (or have a Workaround) against CVE-2021-44228 or CVE-2021-45046 as they use Log4j version 2.x

**Comment by tschmidtb51** [comment]
Feel free to mention @tschmidtb51 once I should have a look again.

**Comment by DFFspace** [comment]
Hi @tschmidtb51 You can have a look again.

**Review by tschmidtb51 — APPROVED** [review]
Congratulations. All changes accepted! :-)

Thank you for this huge contribution!

**Pull request merged**